src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -60,6 +60,10 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                         TokenSource.Cache);
                 }
             }
+            else
```

Comment:
Please raise an issue that "refresh_in" logic is not performed for OBO.

Reply:
Created issue: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2389